File: sorting/binary_search/binary_search.js
const array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

const binarySearch = (array, target) => {
  let low = 0;
  let high = array.length - 1;

  while (low <= high) {
    const mid = Math.floor((low + high) / 2);
    if (array[mid] == target) {
      console.log(`Found ${target} at index ${mid}`);
      return mid;
    } else if (array[mid] < target) {
      low = array[mid] + 1;
    } else {
      high = array[mid] - 1;
    }
  }
  console.log(`${target} not found`);
  return -1;
};

binarySearch(array, 10);

File: sorting/binary_search/binary_search.py
array = [1, 2, 3]
def binary_search(nums, target):
    
    low = 0
    high = len(nums) - 1
    
    while low <= high:

        mid = (high + low) // 2

        # If x is greater, ignore left half
        if nums[mid] < target:
            low = mid + 1

        # If x is smaller, ignore right half
        elif nums[mid] > target:
            high = mid - 1

        # means x is present at mid
        else:
            return mid

    # If we reach here, then the element was not present
    return -1

print(binary_search(array, 2))
